webargs/aiohttpparser.py

```diff
@@ -68,10 +69,14 @@ def parse_json(self, req, name, field):
         if req.has_body and is_json_request(req):
             json_data = self._cache.get('json')
             if json_data is None:
-                self._cache['json'] = yield from req.json()
-            return core.get_value(self._cache['json'], name, field)
-        else:
-            return core.missing
+                self._cache['json'] = json_data = yield from req.json()
+            if json_data is not None:
+                if isinstance(json_data, dict):
```

Comment:
Better to do `isinstance(json_data, collections.Mapping)` to handle all mapping types.